In [4]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression

# Data import

In [12]:
data = pd.read_excel('./RCT_allergyResponse_DataForFinal.xlsx')
data.dropna(inplace = True)
data.head()
data.shape
data.dtypes
data.describe()

subjectNumber  vaccineTreatment  allgergySymptom
0              1                 0                0
1              2                 0                0
2              3                 0                0
3              4                 0                1
4              5                 1                0

(30000, 3)

subjectNumber       int64
vaccineTreatment    int64
allgergySymptom     int64
dtype: object

subjectNumber  vaccineTreatment  allgergySymptom
count   30000.000000      30000.000000     30000.000000
mean    15000.500000          0.503367         0.053933
std      8660.398374          0.499997         0.225890
min         1.000000          0.000000         0.000000
25%      7500.750000          0.000000         0.000000
50%     15000.500000          1.000000         0.000000
75%     22500.250000          1.000000         0.000000
max     30000.000000          1.000000         1.000000

In [15]:
data1 = data[data['vaccineTreatment']==1]
data2 = data[data['vaccineTreatment']==0]

In [16]:
data1.describe()
data2.describe()

subjectNumber  vaccineTreatment  allgergySymptom
count   15101.000000           15101.0     15101.000000
mean    15027.028078               1.0         0.009072
std      8641.557653               0.0         0.094818
min         5.000000               1.0         0.000000
25%      7604.000000               1.0         0.000000
50%     14956.000000               1.0         0.000000
75%     22526.000000               1.0         0.000000
max     30000.000000               1.0         1.000000

subjectNumber  vaccineTreatment  allgergySymptom
count   14899.000000           14899.0     14899.000000
mean    14973.612256               0.0         0.099403
std      8679.660094               0.0         0.299212
min         1.000000               0.0         0.000000
25%      7389.500000               0.0         0.000000
50%     15062.000000               0.0         0.000000
75%     22453.500000               0.0         0.000000
max     29999.000000               0.0         1.000000

# test

In [5]:
y = data['allgergySymptom']
x = data['vaccineTreatment']
lr = LinearRegression()
lr.fit(x.to_numpy().reshape(-1, 1), y)
coef = lr.coef_[0]
intercept = lr.intercept_
residual = y - coef * x - intercept
N_treated = data['vaccineTreatment'].sum()
N = len(data)
N_controlled = N - N_treated
tStat = np.sqrt(N) * coef / np.sqrt( (N/N_treated + N/N_controlled) * residual.var() )
# Number of Controlled_group & Treated_group
print('Number of Controlled_group & Treated_group : %d, %d' %(N_controlled, N_treated))
# The increase due to the vaccine
print('The increase due to the vaccine : %.4f' %coef)
# t-stat of the estimate in b
print('t-stat of the estimate in b : %.4f' %tStat)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

Number of Controlled_group & Treated_group : 14899, 15101
The increase due to the vaccine : -0.0903
t-stat of the estimate in b : -35.3441


t-stat이 -35.3441이다. 이는 귀무가설을 기각하기 때문에, 백신의 주입이 알레르기 증상에 영향을 주지 않는 것을 기각한다는 것이 통계적으로 유의미하게 나타난다고 볼 수 있다.